components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts

```diff
@@ -215,15 +217,24 @@ export class StatsService {
   /* TODO Auth team: Helper functions that are only not private because
      they have unit testing. Should we delete their tests and make
      them private? */
-  getControlStatus(control): string {
-    const statuses = control.results.map(r => r.status);
-    const waived = control.waiver_data && control.waiver_data.skipped_due_to_waiver;
+  getControlStatus(control): ControlStatus {
+    const waived = this.checkIfWaived(control.waived_str);
     if (waived) { return 'waived'; }
+
+    const statuses = control.results.map(r => r.status);
     if (statuses.every(s => s === 'passed')) { return 'passed'; }
+    if (statuses.every(s => s === 'failed')) { return 'failed'; }
```

Comment:
statuses.any ?

any failure == failed

Reply:
yikes - i need to slow down.  I changed to find, so that if it finds a match, its failure.